Patch-mainline: in house
From: Li Yu <bingtian.ly@taobao.com>
Subject: 
References: 

This patch avoid to inline oom_kill_task() function,
then apsaraoom kernel hacks still can steal which is task
to be killed by install a kprobe on it.

Signed-off-by: Li Yu <bingtian.ly@taobao.com>
Acked-by: 

---
diff --git a/mm/oom_kill.c b/mm/oom_kill.c
--- a/mm/oom_kill.c	2013-03-28 15:47:00.867596216 +0800
+++ b/mm/oom_kill.c	2013-03-28 15:46:49.957471671 +0800
@@ -436,7 +436,7 @@ static void dump_header(struct task_stru
 }
 
 #define K(x) ((x) << (PAGE_SHIFT-10))
-static int oom_kill_task(struct task_struct *p, struct mem_cgroup *mem)
+static noinline int oom_kill_task(struct task_struct *p, struct mem_cgroup *mem)
 {
 	struct task_struct *q;
 	struct mm_struct *mm;
